lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/tool/beans/SimplePageBean.java

```diff
@@ -4609,14 +4683,131 @@ private String uploadFile(String collectionId) {
+		List<SimplePageItem> itemsNow = getItemsOnPage(pageId);
+		if (itemsNow==null || itemsNow.isEmpty()){
+			return 1;
+		}
+		int currentHighest = 1;
+		for(SimplePageItem itemNow: itemsNow){
+			if(itemNow.getSequence() > currentHighest){
+				currentHighest = itemNow.getSequence();
+			}
+		}
+		return currentHighest + 1;
+	}
+
+	public String addPageLayout(){
+		int sequence = getNextSequenceOnPage(getCurrentPageId());	//find the starting sequence number based on whatever might already be on the page. Must be incremented after every use.
+       	if(StringUtils.equals(this.pageLayoutSelect, "addSubpageList")){
+			SimplePage pageNow = null;
+			SimplePageItem itemNow = null;
+			if(StringUtils.isBlank(pageSubpageTitle)){
+				setErrMessage(messageLocator.getMessage("simplepage.layout.page.emptyName"));
+				return "failure";
+			}
+			itemNow = simplePageToolDao.makeItem(getCurrentPageId(),sequence,SimplePageItem.BREAK,null,null);
+			sequence++;
+			itemNow.setFormat("section");
+			itemNow.setAttribute("forceBtn","false");
+			itemNow.setAttribute("checklistItems",null);
+			update(itemNow);	//finish writing break before subpages
+			for(int count=1; count<=pageSubpageCount; count++){
+				pageNow = addPage(pageSubpageTitle + ' ' + count, null, false, false);	//make new page
+				pageNow.setParent(getCurrentPageId());	//clean up page data
+				pageNow.setTopParent(getCurrentPage().getTopParent());
+				update(pageNow);	//save revised page data
+				itemNow = simplePageToolDao.findItemsBySakaiId(Long.toString(pageNow.getPageId())).get(0);	//there should only be one item with this sakaiId, because it was just now created as part of addPage.
+				if(pageSubpageButton){	//clean up item data
+					itemNow.setFormat("button");
+				}
+				itemNow.setPageId(getCurrentPageId());
+				itemNow.setAttribute("btnColor",pageButtonColorScheme);
+				itemNow.setSequence(sequence);
+				sequence++;
+				update(itemNow);	//save revised item data
+			}
+		} else if(StringUtils.equals(this.pageLayoutSelect, "interiorResources") || StringUtils.equals(this.pageLayoutSelect, "interiorTask")){	//first handle the parts that these layouts have in common.
+			SimplePageItem itemNow = simplePageToolDao.makeItem(getCurrentPageId(), sequence, SimplePageItem.BREAK, null, null);
+			sequence++;
+			itemNow.setFormat("section");
+			itemNow.setAttribute("colwidth", "2");
+			itemNow.setAttribute("forceBtn","false");
+			itemNow.setAttribute("checklistItems",null);
+			update(itemNow);	//finish writing top Section/break item
+			itemNow = simplePageToolDao.makeItem(getCurrentPageId(), sequence,SimplePageItem.TEXT,null,null);
+			sequence++;
+			String overviewHtmlAppender = "";
+			if(StringUtils.equals(this.pageLayoutSelect, "interiorResources")){	//special Overview text for interiorResources only
+				overviewHtmlAppender = overviewHtmlAppender + messageLocator.getMessage("simplepage.layout.page.overview","Tasks");
```

Comment:
Is this hardcoded in English? I'm fine raising I18N or L10N bugs after merging this.